pyparsing/core.py

```diff
@@ -558,80 +566,32 @@ def postParse(self, instring, loc, tokenlist):
 
     # ~ @profile
     def _parseNoCache(self, instring, loc, doActions=True, callPreParse=True):
-        TRY, MATCH, FAIL = 0, 1, 2
-        debugging = self.debug  # and doActions)
-
-        if debugging or self.failAction:
-            # ~ print("Match", self, "at loc", loc, "(%d, %d)" % (lineno(loc, instring), col(loc, instring)))
-            if self.debugActions[TRY]:
-                self.debugActions[TRY](instring, loc, self)
+        self.debug and self.debugActions.TRY(instring, loc, self)
```

Comment:
You will see this pattern

    self.debug and self.debugActions.TRY(instring, loc, self)

which is the same as

     if self.debug:
        self.debugActions.TRY(instring, loc, self)

Reply:
Not a fan of this pattern.